plone/app/z3cform/converters.py

```diff
@@ -228,7 +228,7 @@ def toWidgetValue(self, value):
             return self.field.missing_value
         separator = getattr(self.widget, 'separator', ';')
         if IRelationList.providedBy(self.field):
-            return separator.join([IUUID(o) for o in value if value])
+            return separator.join([IUUID(o) for o in value if value and o])
```

Comment:
I guess the ``if value`` was a mistake to begin with and they meant ``if o`` as it does not make sense to check on the conditional that the iterable you are iterating over exists... If it does not exist or is empty you don't even reach the conditional:
```
>>> [x for x in []]
[]
>>> range(3)
[0, 1, 2]
>>> [x for x in range(3) if x > 1]
[2]
```